blog/docs/articles/demystifying-activator-on-path.md

```diff
@@ -0,0 +1,182 @@
+result and makes a decision based on the condition `decider.Status.DesiredScale != sRes.DesiredPodCount` whether to trigger a new reconciliation for the KPA reconciler. Goal is the KPA to get the latest scale result.
+
+- creates a Metric resource that triggers the metrics collector controller to setup a scraper for the revision pods.
+
+- calls a scale method that decides the number of wanted pods and also updates the revision deployment
+
+- creates/updates a ServerlessService (SKS) that holds info about the operation mode (proxy or serve) and stores the activators used in proxy mode. That SKS create/update event triggers a reconciliation for the SKS from its specific controller that creates the required public and private K8s services so traffic can be routed to the K8s deployment.
+This in combination with the networking setup done by the net-* components is the
+
+- updates the PA and reports the active and wanted pods in its status
+
+## Capacity and Operation Modes in Practice
+
+As described earlier Activator will be removed if enough capacity is available and there is an invariant that needs to
+hold, that is EBC (excess burst capacity)>0, where EBC = TotalCapacity - ObservedInPanicMode - TargetBurstCapacity(TBC).
+
+Let's see an example of a service that has a target concurrency of 10 and tbc=10:
+
+```
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: autoscale-go
+spec:
+  template:
+    metadata:
+      annotations:
+        autoscaling.knative.dev/target: "10"
+        autoscaling.knative.dev/target-burst-capacity: "10"
+    spec:
+      containers:
+      - image: ghcr.io/knative/autoscale-go:latest
+```
+
+Initially when the ksvc was deployed there was no traffic and one pod is created by default for verification reasons.
+
+Until the pod is up we have:
+```bash
+$ kubectl get sks
+NAME                 MODE    ACTIVATORS   SERVICENAME          PRIVATESERVICENAME           READY     REASON
+autoscale-go-00001   Proxy   2            autoscale-go-00001   autoscale-go-00001-private   Unknown   NoHealthyBackends
+```
+When the pod is up we have:
+
+```bash
+$ oc get po
+NAME                                             READY   STATUS    RESTARTS   AGE
+autoscale-go-00001-deployment-6cc679b9d6-xgrkf   2/2     Running   0          24s
+
+$ oc get sks
+NAME                 MODE    ACTIVATORS   SERVICENAME          PRIVATESERVICENAME           READY   REASON
+autoscale-go-00001   Serve   2            autoscale-go-00001   autoscale-go-00001-private   True    
+```
+
+The reason why we are in Serve mode is because EBC=0. In the logs we get:
+
+
+```bash
+{"severity":"DEBUG","timestamp":"2023-10-10T15:29:37.241575214Z","logger":"autoscaler","caller":"scaling/autoscaler.go:286","message":"PodCount=1 Total1PodCapacity=10.000 ObsStableValue=0.000 ObsPanicValue=0.000 TargetBC=10.000 ExcessBC=0.000","commit":"f1617ef","knative.dev/key":"default/autoscale-go-00001"}
```

Comment:
Maybe extract the message bits from the JSON log or format the json. its a bit hard to read in a blog-post.
Same for the log lines below.

Reply:
Yes it was mostly dumped here for reference.